### PR TITLE
Tests for model versions table and empty state

### PR DESCRIPTION
**Pull request by manaswinidas**
Tests for https://issues.redhat.com/browse/RHOAIENG-2236
 https://issues.redhat.com/browse/RHOAIENG-2237
 https://issues.redhat.com/browse/RHOAIENG-2234

## Description

 - Added e2e tests for model versions empty state and model versions table
 - Fixed Actions button from Split button to normal dropdown according to mocks 
 - The chip key values now reflects owner instead of just keyword
 - Fixed the `app-page-title` along with tests for the same.

## How Has This Been Tested?

Run Cypress test -`ModelRegistry.cy.ts`

## Request review criteria:

Self checklist (all need to be checked):
- [X] The developer has manually tested the changes and verified that the changes work
- [X] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [X] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [X] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: frontend/src/__mocks__/mockModelVersionList.ts [hunk 1]
```diff
@@ -1,9 +1,11 @@
...
-export const mockModelVersionList = (): ModelVersionList => ({
-  items: [mockModelVersion({ author: 'Author 1', registeredModelId: '1' })],
+export const mockModelVersionList = ({
+  items = [],
+}: Partial<ModelVersionList>): ModelVersionList => ({
+  items,
   nextPageToken: '',
   pageSize: 0,
-  size: 1,
+  size: 2,
```
Comment (dpanshug) [inline frontend/src/__mocks__/mockModelVersionList.ts:10]: make the size dynamic
```suggestion
  size: items.length,
```

File: frontend/src/__mocks__/mockModelVersion.ts [hunk 2]
```diff
@@ -1,20 +1,24 @@
...
   author = 'Test author',
   registeredModelId = '1',
+  name = 'new model version',
+  customProperties = {},
 }: MockModelVersionType): ModelVersion => ({
   author,
   createTimeSinceEpoch: '1712234877179',
-  customProperties: {},
+  customProperties,
   id: '26',
```
Comment (dpanshug) [inline frontend/src/__mocks__/mockModelVersion.ts:19]: can we make id dynamic too? as we are mocking multiple model versions

File: frontend/src/pages/modelRegistry/screens/ModelRegistry.tsx [hunk 3]
```diff
@@ -30,7 +31,7 @@ const ModelRegistry: React.FC = () => {
         />
       }
       title={
-        <TitleWithIcon title="Deployed models" objectType={ProjectObjectType.deployedModels} />
+        <TitleWithIcon title="Registered models" objectType={ProjectObjectType.deployedModels} />
```
Comment (dpanshug) [inline frontend/src/pages/modelRegistry/screens/ModelRegistry.tsx:34]: @lucferbux this was previously changed to "Deployed models", can you check this?

**Review by dpanshug — CHANGES_REQUESTED** [review]
Just some small changes, and also can you pls move Model version tests to another file?

File: frontend/src/__tests__/cypress/cypress/e2e/modelRegistry/ModelRegistry.cy.ts
Comment (dpanshug) [inline frontend/src/__tests__/cypress/cypress/e2e/modelRegistry/ModelRegistry.cy.ts:1]: After moving the model version tests to different file, does this file needs the rest of the modifications?

File: frontend/src/pages/modelRegistry/screens/ModelRegistry.tsx (hunk 3 above)
Comment (lucferbux) [reply frontend/src/pages/modelRegistry/screens/ModelRegistry.tsx:34]: It was a mistake, @manaswinidas did the right thing here

File: frontend/src/__tests__/cypress/cypress/e2e/modelRegistry/ModelRegistry.cy.ts
Comment (lucferbux) [reply frontend/src/__tests__/cypress/cypress/e2e/modelRegistry/ModelRegistry.cy.ts:1]: Model version is no longer needed, but we can refactor later

**Review by lucferbux — APPROVED** [review]
/lgtm

Seems great, there's a bit of a refactor in model registry view in cypress that we'll need to perform

**Pull request merged**